workers/unity/Packages/com.improbable.gdk.health/Systems/HealthRegenSystem.cs

```diff
@@ -16,38 +15,45 @@ public struct EntitiesNeedingRegenData
             [ReadOnly] public EntityArray Entities;
             [ReadOnly] public ComponentDataArray<HealthRegenComponent.Component> HealthRegenComponents;
             [ReadOnly] public SubtractiveComponent<HealthRegenData> DenotesMissingData;
-            [ReadOnly] public ComponentDataArray<Authoritative<HealthComponent.Component>> DenotesAuthority;
+            [ReadOnly] public SharedComponentDataArray<HealthComponent.ComponentAuthority> Authority;
         }
 
         public struct TakingDamage
         {
             public readonly int Length;
             public ComponentDataArray<HealthRegenData> RegenData;
             public ComponentDataArray<HealthRegenComponent.Component> HealthRegenComponents;
-            [ReadOnly] public ComponentDataArray<HealthComponent.ReceivedEvents.HealthModified> HealthModifiedEvents;
-            [ReadOnly] public ComponentDataArray<Authoritative<HealthComponent.Component>> DenotesAuthority;
+            [ReadOnly] public ComponentDataArray<SpatialEntityId> EntityIds;
+            [ReadOnly] public SharedComponentDataArray<HealthComponent.ComponentAuthority> Authority;
         }
 
         public struct EntitiesToRegen
         {
             public readonly int Length;
-            public ComponentDataArray<HealthComponent.CommandSenders.ModifyHealth> ModifyHealthCommandSenders;
             public ComponentDataArray<HealthRegenComponent.Component> HealthRegenComponents;
             public ComponentDataArray<HealthRegenData> RegenData;
             [ReadOnly] public ComponentDataArray<HealthComponent.Component> HealthComponents;
             [ReadOnly] public ComponentDataArray<SpatialEntityId> EntityId;
-            [ReadOnly] public ComponentDataArray<Authoritative<HealthComponent.Component>> DenotesAuthority;
+            [ReadOnly] public SharedComponentDataArray<HealthComponent.ComponentAuthority> Authority;
+            [ReadOnly] public EntityArray Entities;
         }
 
         [Inject] private EntitiesNeedingRegenData needData;
         [Inject] private TakingDamage takingDamage;
         [Inject] private EntitiesToRegen toRegen;
+        [Inject] private CommandSystem commandSystem;
+        [Inject] private ComponentUpdateSystem componentUpdateSystem;
 
         protected override void OnUpdate()
         {
             // Add the HealthRegenData if you don't currently have it.
             for (var i = 0; i < needData.Length; i++)
             {
+                if (!needData.Authority[i].HasAuthority)
```

Comment:
This should use SetFilter instead.
This require's us to no longer `Inject` structs but to make proper groups, which is the future API post Entities.26 anyways.

Reply:
This was the minimal change needed to test the difference